src/test/java/net/sf/jabref/logic/autocompleter/BibtexKeyAutoCompleterTest.java

```diff
@@ -0,0 +1,153 @@
+package net.sf.jabref.logic.autocompleter;
+
+import static org.mockito.Mockito.*;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sf.jabref.model.entry.BibtexEntry;
+
+public class BibtexKeyAutoCompleterTest {
+
+    @Test(expected = NullPointerException.class)
+    public void initAutoCompleterWithNullPreferenceThrowsException() {
+        new BibtexKeyAutoCompleter(null);
+    }
+
+    @Test
+    public void completeWithoutAddingAnythingReturnsNothing() {
+        AutoCompletePreferences preferences = mock(AutoCompletePreferences.class);
+        BibtexKeyAutoCompleter autoCompleter = new BibtexKeyAutoCompleter(preferences);
+
+        String[] result = autoCompleter.complete("test");
+        Assert.assertEquals(0, result.length);
```

Comment:
Mhhh... do you really find the test failure message `AssertionError: array lengths differed, expected.length=0 actual.length=1` more expressive? I have more problems understanding the line `Assert.assertArrayEquals(new String[] {}, result);`. 

Maybe I should just replace the array by an `ArrayList`, then one could simply check `isEmpty()`. Best solution?